src/app/content/highlights/components/SummaryPopup/HighlightListElement.tsx

```diff
@@ -77,7 +78,16 @@ const HighlightListElement = ({ highlight, locationFilterId, pageId }: Highlight
   const [isDeleting, setIsDeleting] = React.useState(false);
   const book = useSelector(bookSelector);
   const dispatch = useDispatch();
-  const linkToHighlight = React.useMemo(() => createHighlightLink(highlight, book), [highlight, book]);
+  const services = useServices();
+  const [linkToHighlight, setLinkToHighlight] = useState('');
+
+  React.useEffect(() => {
+    const getLinkToHighlight = async() => {
+      const highLightLink = await createHighlightLink(services.bookConfigLoader, highlight, book);
+      setLinkToHighlight(highLightLink);
+    };
+    getLinkToHighlight();
+  }, [book, highlight, services.bookConfigLoader]);
```

Comment:
i feel like a `useContentLink(book, page)` hook might be in order, although i see that is not exactly what you're doing here, maybe you could compose it and still have less logic